docs/style-guide.md

```diff
@@ -199,3 +201,22 @@ We could enforce a camel case convention on these strings, such as the following
 ##### Decision
 
 - Defer the decision on the syntax for identifiers to the other specification when possible.
+
+## Element Ordering
+
+This section concerns the order in which containers store elements. 
+
+### General Guidelines
+
+:star2: ECMA-402 spec must provide a deterministic order for the contents of all containers. This order should be lexicographic except in cases wherein there is a clearly preferable semantic ordering. :star2:
+  
+For an example of when *not* to use lexicographic order, consider an array holding days of the week. This array should store those days in the order `["monday", "tuesday", "wednesday", ...]` instead of the lexicographic `["friday", "monday", "saturday", "sunday", ...]` order.
+
+### `resolvedOptions`
+
+:star2: The `resolvedOptions` of Intl objects should appear in the following order: 
+
+1. `locale`
+2. All properties (given in lexicographical order) that can be set by extension keys and that are guaranteed to exist
+3. Properties (in lexicographical order) that are not set by extension keys and that are guaranteed to exist 
+4. All properties (in lexicographical order) that exist conditionally. :star2:
```

Comment:
@gibson042 for feedback on this ordering

Reply:
It seems insufficiently stable to me, because we are currently contemplating adding support for -u- extension keys that can currently only be set by options, and could definitely promote a conditional property to mandatory in the future.

```suggestion
### `resolvedOptions`

:star2: *Properties in the output from `resolvedOptions()` of an Intl object should start with `locale` and otherwise use lexicographic order. Deviations (such as those motivated by semantic order involving a subset of properties) must be documented.* :star2:

Spec changes that add properties should do so in accordance with this recommendation, rather than automatically placing them at the end—relative enumeration order of properties should remain stable over time, but there is no such expectation regarding adjacency.

#### Examples

`Object.keys(new Intl.Segmenter().resolvedOptions())` returns `[ 'locale', 'granularity' ]`. If support for the [Unicode BCP 47 U Extension "dx" key](https://unicode.org/reports/tr35/#UnicodeDictionaryBreakExclusionIdentifier) were added and exposed as `dictionaryBreakExcludedScripts`, that property would belong before `granularity`.
```